streams/shared/src/main/scala/zio/stream/ZPipeline.scala

```diff
@@ -2078,139 +2078,204 @@ object ZPipeline extends ZPipelinePlatformSpecificConstructors {
    */
   def splitOn(delimiter: => String)(implicit trace: Trace): ZPipeline[Any, Nothing, String, String] =
     ZPipeline.mapChunks[String, Char](_.flatMap(string => Chunk.fromArray(string.toArray))) >>>
-      ZPipeline.splitOnChunk[Char](Chunk.fromArray(delimiter.toArray)) >>>
+      ZPipeline.splitOnChunk[Char, Char](delimiter.toList, true) >>>
       ZPipeline.mapChunks[Char, String](chunk => Chunk.single(chunk.mkString("")))
 
   /**
-   * Splits strings on a delimiter.
+   * Splits strings on several possible delimiters.
    */
-  def splitOnChunk[In](delimiter: => Chunk[In])(implicit trace: Trace): ZPipeline[Any, Nothing, In, In] =
-    ZPipeline.suspend {
+  def splitOnMany(delimiters: => Seq[String])(implicit trace: Trace): ZPipeline[Any, Nothing, String, String] =
+    ZPipeline.mapChunks[String, Char](_.flatMap(string => Chunk.fromArray(string.toArray))) >>>
+      ZPipeline.splitOnManyChunk[Char, Char](delimiters.map(_.toList), true) >>>
+      ZPipeline.mapChunks[Char, String](chunk => Chunk.single(chunk.mkString("")))
 
-      def next(
-        leftover: Option[Chunk[In]],
-        delimiterIndex: Int
-      ): ZChannel[Any, ZNothing, Chunk[In], Any, Nothing, Chunk[In], Any] =
+  private final case class Trie[T1](structure: Map[T1, Trie[T1]], depth: Int, isLeaf: Boolean = false) {
+    def partitionAll[T <: T1](
+      chunk: Chunk[T],
+      carryTries: Seq[Trie[T1]],
+      offset: Int,
+      allowEmpty: Boolean = true
+    ): ((Chunk[T], Seq[Trie[T1]]), Iterable[Chunk[T]]) = {
+      val iterator          = chunk.chunkIterator
+      var partitionBuffer   = scala.collection.mutable.ListBuffer.empty[Chunk[T]]
+      var curTries          = carryTries
+      var index             = offset
+      var curPartitionStart = 0
+      while (iterator.hasNextAt(index)) {
+        curTries = curTries.flatMap(_.structure.get(iterator.nextAt(index)))
+        index += 1
+        curTries.filter(_.isLeaf).reverse.headOption match {
+          case Some(leaf) => {
+            if (index - leaf.depth > curPartitionStart || allowEmpty) {
+              partitionBuffer += chunk.slice(curPartitionStart, index - leaf.depth)
+            }
+            curPartitionStart = index
+            curTries = Seq.empty
+          }
+          case None => {}
+        }
+        curTries = this +: curTries
+      }
+      ((chunk.takeRight(chunk.length - curPartitionStart), curTries), partitionBuffer)
+    }
+  }
+
+  private object Trie {
+    def empty[T]            = Trie(Map.empty[T, Trie[T]], 0, false)
```

Comment:
Little trick used in ZIO codebase to avoid allocating more than one `empty`

```suggestion
    private val _empty               = Trie(structure = Map.empty[T, Trie[T]], depth = 0, isLeaf = false)
    def empty[T]: Trie[T]            = _empty.asInstanceOf[Trie[T]]
```